prow/config/config.go

```diff
@@ -527,6 +529,26 @@ func yamlToConfig(path string, nc interface{}) error {
 	return nil
 }
 
+// ReadFileMaybeGZIP wraps ioutil.ReadFile, returning the decompressing contents
+// if the file is gzipped, or otherwise the raw contents
+func ReadFileMaybeGZIP(path string) ([]byte, error) {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	// check if file contains gzip header: http://www.zlib.org/rfc-gzip.html
+	if !bytes.HasPrefix(b, []byte("\x1F\x8B\x08")) {
```

Comment:
NOTE: `net/http.DetectContentType` [uses this same pattern](https://golang.org/src/net/http/sniff.go#L147). *Technically* this is GZIP/deflate, but the other options are unlikely. We could drop `\x08` for just GZIP.

```
  0      2 bytes  magic header  0x1f, 0x8b (\037 \213)  
  2      1 byte   compression method
                     0: store (copied)
                     1: compress
                     2: pack
                     3: lzh
                     4..7: reserved
                     8: deflate
```
http://www.onicos.com/staff/iz/formats/gzip.html

Reply:
right, one more example https://github.com/google/gopacket/blob/master/pcapgo/read.go#L52-L53 

+1 to drop 0x08

Reply:
done 👍